src/OpenDiffix.Core/Anonymizer.fs

```diff
@@ -47,25 +47,39 @@ let private generateNoise salt stepName stdDev noiseLayers =
 
 // Compacts flattening intervals to fit into the total count of contributors.
 // Both intervals are reduced proportionally, with `topCount` taking priority.
+// NOTE `totalCount >= outlierCount.Lower + topCount.Lower` is assumed
```

Comment:
It is better to put stuff like this in an assert rather than in a comment.

Reply:
it kinda is there, in the `Not possible` branch, but agreed, we could trade it for something that is put here. I'll think about this

Reply:
Can be the first statement in the function. F# supports [assert](https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/assertions). We should utilize it more!

Reply:
:+1: I'm giving this bit a revamp, but I'll `assert` if still relevant afterwards